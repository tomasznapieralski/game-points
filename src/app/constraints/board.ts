export const BOARD_DIMENSIONS = {
  ROWS: 5,
  COLS: 5,
};
