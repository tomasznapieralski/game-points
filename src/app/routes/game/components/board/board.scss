.board {
  display: flex;
  flex-direction: column;
  height: 100%;

  &__title {
    display: flex;
    align-items: center;
    height: 60px;
    padding: 10px 20px;
    border-bottom: 2px solid #885900;
  }

  &__logo {
    height: 35px;
  }

  &__subtitle {
    margin-top: 8px;
    margin-left: 2px;
    font-size: 20px;
    text-transform: uppercase;
    letter-spacing: 2px;
  }

  &__playground {
    position: relative;
    flex-grow: 1;
    padding: 10px;
  }

  &__playground-item {
    position: absolute;
    width: 65px;
    height: 65px;
    font-size: 60px;
    transition: all 0.3s ease;

    &--hidden {
      transform: scale(0);
    }

    @for $row from 1 through 5 {
      &--row-#{$row} {
        top: ($row - 1) * 100px + 35px;
      }
    }

    @for $col from 1 through 5 {
      &--col-#{$col} {
        left: ($col - 1) * 110px + 25px;
      }
    }
  }
}
