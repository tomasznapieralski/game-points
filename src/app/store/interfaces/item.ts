export interface ItemInterface {
  icon: string;
  basePoints: number;
  bonusPoints: number;
  bonusMultiplier: number;
}

export interface CollectedItemInterface {
  item: ItemInterface;
  quantity: number;
}
