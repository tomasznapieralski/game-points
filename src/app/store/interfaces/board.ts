import { ItemInterface } from './item';

export interface BoardSlotInterface {
  id: number;
  row: number;
  col: number;
  isHidden: boolean;
  item: ItemInterface | null;
}
