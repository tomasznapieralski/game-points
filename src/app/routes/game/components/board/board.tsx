import React from 'react';
import { Action } from 'redux';
import classNames from 'classnames';

import Item from '../item/item';

import logo from '../../../../../assets/logo.png';

import { BoardSlotInterface } from '../../../../store/interfaces/board';

import './board.scss';

interface PropsInterface {
  boardSlots: BoardSlotInterface[];
  handleBoardItemClick: (id: number) => Action;
  handleBoardItemInit: (id: number) => Action;
  handleBoardItemTimeout: (id: number) => Action;
}

const Board: React.FC<PropsInterface> = ({
  boardSlots,
  handleBoardItemClick,
  handleBoardItemInit,
  handleBoardItemTimeout,
}) => {
  return (
    <div className="board">
      <div className="board__title">
        <img
          className="board__logo"
          src={logo}
          alt=""
        />
        <div className="board__subtitle">
          Points
        </div>
      </div>
      <div className="board__playground">
        {boardSlots && boardSlots.map(({
          row,
          col,
          id,
          isHidden,
          item,
        }) =>
          <div
            className={classNames(
              'board__playground-item',
              `board__playground-item--row-${row}`,
              `board__playground-item--col-${col}`,
              {
                'board__playground-item--hidden': isHidden,
              }
            )}
            key={id}
          >
            {item && <Item
              icon={item.icon}
              id={id}
              initHandler={handleBoardItemInit}
              timeoutHandler={handleBoardItemTimeout}
              clickHandler={() => handleBoardItemClick(id)}
            />}
          </div>
        )}
      </div>
    </div>
  );
}

export default Board;
