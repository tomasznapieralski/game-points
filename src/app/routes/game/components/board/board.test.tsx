import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';

import { BoardSlotInterface } from '../../../../store/interfaces/board';

import Board from './board';

afterEach(cleanup);

test('Board displays the items with correct classes', () => {
  const boardSlots: BoardSlotInterface[] = [
    {
      id: 11,
      row: 1,
      col: 1,
      isHidden: true,
      item: null,
    },
    {
      id: 22,
      row: 2,
      col: 2,
      isHidden: false,
      item: {
        icon: 'fish',
        basePoints: 150,
        bonusPoints: 100,
        bonusMultiplier: 3,
      },
    },
    {
      id: 33,
      row: 3,
      col: 3,
      isHidden: true,
      item: {
        icon: 'fish',
        basePoints: 150,
        bonusPoints: 100,
        bonusMultiplier: 3,
      },
    },
  ];

  const tree = renderer
    .create(
      <Board
        boardSlots={boardSlots}
        handleBoardItemClick={jest.fn()}
        handleBoardItemInit={jest.fn()}
        handleBoardItemTimeout={jest.fn()}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();

  const { container } = render(
    <Board
      boardSlots={boardSlots}
      handleBoardItemClick={jest.fn()}
      handleBoardItemInit={jest.fn()}
      handleBoardItemTimeout={jest.fn()}
    />
  );

  expect(container.getElementsByClassName('board__playground-item')).toHaveLength(3);
  expect(container.getElementsByClassName('board__playground-item--hidden')).toHaveLength(2);
  expect(container.getElementsByClassName('board__playground-item--row-1')[0].className)
    .toBe('board__playground-item board__playground-item--row-1 board__playground-item--col-1 board__playground-item--hidden');
  expect(container.getElementsByClassName('board__playground-item--row-2')[0].className)
    .toBe('board__playground-item board__playground-item--row-2 board__playground-item--col-2');
  expect(container.getElementsByClassName('board__playground-item--row-3')[0].className)
    .toBe('board__playground-item board__playground-item--row-3 board__playground-item--col-3 board__playground-item--hidden');
});
