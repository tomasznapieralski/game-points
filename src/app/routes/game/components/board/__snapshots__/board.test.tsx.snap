// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board displays the items with correct classes 1`] = `
<div
  className="board"
>
  <div
    className="board__title"
  >
    <img
      alt=""
      className="board__logo"
      src="logo.png"
    />
    <div
      className="board__subtitle"
    >
      Points
    </div>
  </div>
  <div
    className="board__playground"
  >
    <div
      className="board__playground-item board__playground-item--row-1 board__playground-item--col-1 board__playground-item--hidden"
    />
    <div
      className="board__playground-item board__playground-item--row-2 board__playground-item--col-2"
    >
      <div
        className="item item--clickable"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-fish fa-w-18 item__icon item__icon--fish"
          data-icon="fish"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M327.1 96c-89.97 0-168.54 54.77-212.27 101.63L27.5 131.58c-12.13-9.18-30.24.6-27.14 14.66L24.54 256 .35 365.77c-3.1 14.06 15.01 23.83 27.14 14.66l87.33-66.05C158.55 361.23 237.13 416 327.1 416 464.56 416 576 288 576 256S464.56 96 327.1 96zm87.43 184c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24 13.26 0 24 10.74 24 24 0 13.25-10.75 24-24 24z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
    <div
      className="board__playground-item board__playground-item--row-3 board__playground-item--col-3 board__playground-item--hidden"
    >
      <div
        className="item item--clickable"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-fish fa-w-18 item__icon item__icon--fish"
          data-icon="fish"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M327.1 96c-89.97 0-168.54 54.77-212.27 101.63L27.5 131.58c-12.13-9.18-30.24.6-27.14 14.66L24.54 256 .35 365.77c-3.1 14.06 15.01 23.83 27.14 14.66l87.33-66.05C158.55 361.23 237.13 416 327.1 416 464.56 416 576 288 576 256S464.56 96 327.1 96zm87.43 184c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24 13.26 0 24 10.74 24 24 0 13.25-10.75 24-24 24z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
